plugins/woocommerce/src/Internal/EmailEditor/EmailTemplates/TemplateApiController.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Internal\EmailEditor\EmailTemplates;
+
+use Automattic\WooCommerce\EmailEditor\Validator\Builder;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * API Controller for managing WooCommerce email templates via extending the post type API.
+ *
+ * @internal
+ */
+class TemplateApiController {
+	/**
+	 * Returns the sender settings for the given template.
+	 *
+	 * @param array $template_data - WP_Block_Template data.
+	 * @return array
+	 */
+	public function get_template_data( $template_data ): array {
+		$template_slug = $template_data['slug'] ?? null;
+		if ( WooEmailTemplate::TEMPLATE_SLUG !== $template_slug ) {
+			return array();
+		}
+
+		return array(
+			'sender_settings' => array(
+				'from_name'    => get_option( 'woocommerce_email_from_name' ),
+				'from_address' => get_option( 'woocommerce_email_from_address' ),
+			),
+		);
+	}
+
+	/**
+	 * Update WooCommerce specific data we store with Template.
+	 *
+	 * @param array              $data - WP_Block_Template data.
+	 * @param \WP_Block_Template $template_post - WP_Block_Template object.
+	 */
+	public function save_template_data( array $data, \WP_Block_Template $template_post ): void {
+		if ( WooEmailTemplate::TEMPLATE_SLUG === $template_post->slug && isset( $data['sender_settings'] ) ) {
+			update_option( 'woocommerce_email_from_name', $data['sender_settings']['from_name'] );
+			update_option( 'woocommerce_email_from_address', $data['sender_settings']['from_address'] );
+		}
```

Comment:
Just out of curiosity. I see we don't check the `from_name` key is present in `$data['sender_settings']['from_name']`. Does the API do this via the schema? I assume it does, but I'm not sure.

Reply:
This is good point. The schema does not check it because the endpoint should update data only for the specific template only. I will add a code with some checks.